### DISCUSSION TITLE
Catch all events from client side

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1161] [Q&A] [answered]
```
class MyCustomNamespace(socketio.Namespace):
    def trigger_event(self, event_name, sid, *args):
        print(f"{event_name=}, {sid=}")
        if args:
            print(f"data is {args[0]}")


sio.register_namespace(MyCustomNamespace())
```

You sent me this code  and j couldn't use it because I didn't understand it.. I inputted it into my code and it raised an error 
`ValueError: Not a namespace instance` please could you explain how to use this correctly 

On client version 4.6.1

**Comment by miguelgrinberg** [comment, marked as answer]
The use of namespaces is documented, you can learn this on your own. Here I copy-pasted an example that uses a server namespace class. You are using the client, so you need the ClientNamespace class instead.
